statsmodels/tsa/arima/estimators/tests/test_hannan_rissanen.py

```diff
@@ -100,3 +106,190 @@ def test_unbiased_error():
+            free_ar_ix=np.array([], dtype=int),
+            free_ma_ix=np.array([], dtype=int),
+            # fixed params
+            fixed_ar_params=np.array([]), fixed_ma_params=np.array([0.1, 0.2]),
+        )),
+    ]
+)
+def test_package_fixed_and_free_params_info(fixed_params, spec_ar_lags,
+                                            spec_ma_lags, expected_bunch):
+    actual_bunch = _package_fixed_and_free_params_info(
+        fixed_params, spec_ar_lags, spec_ma_lags
+    )
+    assert isinstance(actual_bunch, Bunch)
+    assert len(actual_bunch) == len(expected_bunch)
+    assert actual_bunch.keys() == expected_bunch.keys()
+
+    # check lags
+    lags = ['fixed_ar_lags', 'fixed_ma_lags', 'free_ar_lags', 'free_ma_lags']
+    for k in lags:
+        assert isinstance(actual_bunch[k], list)
+        assert actual_bunch[k] == expected_bunch[k]
+
+    # check lags
+    ixs = ['fixed_ar_ix', 'fixed_ma_ix', 'free_ar_ix', 'free_ma_ix']
+    for k in ixs:
+        assert isinstance(actual_bunch[k], np.ndarray)
+        assert actual_bunch[k].dtype in [np.int64, np.int32]
+        np.testing.assert_array_equal(actual_bunch[k], expected_bunch[k])
+
+    params = ['fixed_ar_params', 'fixed_ma_params']
+    for k in params:
+        assert isinstance(actual_bunch[k], np.ndarray)
+        np.testing.assert_array_equal(actual_bunch[k], expected_bunch[k])
+
+
+@pytest.mark.parametrize(
+    "fixed_lags, free_lags, fixed_params, free_params, expected_all_params",
+    [
+        ([], [], [], [], []),
+        ([2], [], [0.2], [], [0.2]),
+        ([], [1], [], [0.2], [0.2]),
+        ([1], [3], [0.2], [-0.2], [0.2, -0.2]),
+        ([3], [1, 2], [0.2], [0.3, -0.2], [0.3, -0.2, 0.2]),
+        ([3, 1], [2, 4], [0.3, 0.1], [0.5, 0.], [0.1, 0.5, 0.3, 0.]),
+        ([3, 10], [1, 2], [0.2, 0.5], [0.3, -0.2], [0.3, -0.2, 0.2, 0.5]),
+    ]
+)
+def test_stitch_fixed_and_free_params(fixed_lags, free_lags, fixed_params,
+                                      free_params, expected_all_params):
+    actual_all_params = _stitch_fixed_and_free_params(
+        fixed_lags, fixed_params, free_lags, free_params
+    )
+    assert actual_all_params == expected_all_params
+
+
+def test_unbiased_error_with_fixed_params():
+    # unbiased=True or unbiased=None with fixed params should throw
+    # NotImplementedError for now
+    endog = np.random.normal(size=1000)
+    with pytest.raises(NotImplementedError):
+        hannan_rissanen(endog, ar_order=1, ma_order=1, unbiased=True,
+                        fixed_params={"ar.L1": 0})
+    with pytest.raises(NotImplementedError):
+        hannan_rissanen(endog, ar_order=1, ma_order=1, unbiased=None,
+                        fixed_params={"ar.L1": 0})
+
+
+@pytest.mark.parametrize(
+    "fixed_params",
+    [
+        {"ar.L1": 0.69607715},  # fix ar
+        {"ma.L1": 0.37879692},  # fix ma
+        {"ar.L1": 0.69607715, "ma.L1": 0.37879692},  # no free params
+    ]
+)
+def test_itsmr_with_fixed_params(fixed_params):
```

Comment:
Here are a few examples of fixing 1 or more parameters for Example 5.1.7 in Brockwell and Davis (2016).